site/src/pages/TasksPage/TasksPage.tsx

```diff
@@ -96,10 +110,56 @@ const TasksPage: FC = () => {
 		(t) => t.workspace_id !== null,
 	).length;
 
+	// Fetch notification preferences and templates
+	const [disabledPreferencesQuery, systemTemplatesQuery] = useQueries({
+		queries: [
+			{
+				...userNotificationPreferences(user.id),
+				select: selectDisabledPreferences,
+			},
+			systemNotificationTemplates(),
+		],
+	});
+
+	const disabledPreferences = disabledPreferencesQuery.data ?? {};
+
+	// Check if ALL task notifications are disabled
+	// Returns true only when all task notification templates are disabled.
+	// If even one is enabled, returns false and the warning won't show.
+	const allTaskNotificationsDisabled = systemTemplatesQuery.data
+		?.filter(isTaskNotification)
+		.every((template) => notificationIsDisabled(disabledPreferences, template));
+
+	const queryClient = useQueryClient();
+	const preferencesQuery = useQuery(preferenceSettings());
+	const updatePreferencesMutation = useMutation(
+		updatePreferenceSettings(queryClient),
+	);
+
+	const alertDismissed =
```

Comment:
Nit: Variable name could be more specific.

Reply:
Addressed in https://github.com/coder/coder/pull/20900/commits/284a6514876069cc1ed3e8400fa05587dc10360a